### PR TITLE
Actualize vault TLS flags

### PR DESCRIPTION
**Pull request by Zhaars**
Actualize Vault TLS flags according to this [PR](https://github.com/cossacklabs/acra/pull/578)

### CODE REVIEW

File: content/acra/configuring-maintaining/general-configuration/acra-addzone.md [hunk 1]
```diff
@@ -224,42 +224,142 @@ Should be provided only with `--keystore_encryption_type=<kms_encrypted_master_k
...
+* `--vault_tls_client_cert=<filename>`
+
+  Path to AcraServer TLS certificate presented to Vault (AcraServer works as "client" when communicating with Vault).
+  Empty by default.
+
+
+* `--vault_tls_client_key=<filename>`
 
-  Path to CA certificate bundle to use for HashiCorp Vault certificate validation.
+  Path to private key of the TLS certificate presented to Vault.
```
Comment (vixentael) [inline content/acra/configuring-maintaining/general-configuration/acra-addzone.md:272]:
```suggestion
  Path to AcraServer TLS certificate's private key of the TLS certificate presented to Vault (AcraServer works as "client" when communicating with Vault).
```

yes? it's a private key from AS certificate?

File: content/acra/configuring-maintaining/general-configuration/acra-addzone.md [hunk 2]
```diff
@@ -224,42 +224,142 @@ Should be provided only with `--keystore_encryption_type=<kms_encrypted_master_k
...
+  * `1` — request client certificate, but don't require it;
+  * `2` — expect to receive at least one certificate to continue the handshake;
+  * `3` — don't require client certificate, but validate it if client actually sent it;
+  * `4` — (default) request and validate client certificate.
+
+  These values correspond to [crypto.tls.ClientAuthType](https://golang.org/pkg/crypto/tls/#ClientAuthType).
+
+* `--vault_tls_client_ca=<filename>`
+
+  Path to additional CA certificate for Vault certificate validation.
```
Comment (vixentael) [inline content/acra/configuring-maintaining/general-configuration/acra-addzone.md:260]:
```suggestion
  Path to AcraServer TLS certificate's CA certificate for Vault certificate validation (AcraServer works as "client" when communicating with Vault).
```

right? here acraserver is a client

**Review by vixentael — CHANGES_REQUESTED** [review]
i like how clear the docs become!

pls take a look on few suggestions

File: content/acra/configuring-maintaining/general-configuration/acra-addzone.md [hunk 3]
```diff
@@ -224,42 +224,142 @@ Should be provided only with `--keystore_encryption_type=<kms_encrypted_master_k
...
+  Default is empty (`ACRA_MASTER_KEY` environment variable is expected).
 
-* `--vault_secrets_path=<kv-path>`
+* `--vault_secrets_path=<path>`
 
-  Path to KV Secrets directory in Vault used to store `ACRA_MASTER_KEY`.
+  KV Secret Path for reading `ACRA_MASTER_KEY` from HashiCorp Vault.
   Default is `secret/`.
 
-* `--vault_tls_ca_path=<path>`
```
Comment (Lagovas) [inline content/acra/configuring-maintaining/general-configuration/acra-addzone.md:241]: we should leave it too due to keeping backward compatibility, as commented in related pr

File: content/acra/configuring-maintaining/general-configuration/acra-authmanager.md [hunk 4]
```diff
@@ -71,42 +71,148 @@ read an encrypted file with authentication data.
...
+
+* `--vault_tls_ca_path=<filename>`
+
+  Path to CA certificate for HashiCorp Vault certificate validation.
+  Default is empty (deprecated since 0.94.0, use `vault_tls_client_ca` instead).
+
+
+* `--vault_tls_client_ca=<filename>`
+
+  Path to additional CA certificate for Vault certificate validation.
```
Comment (vixentael) [inline content/acra/configuring-maintaining/general-configuration/acra-authmanager.md:113]: @Zhaars could you please change all `--vault_tls_client*` fields with more clarifing wording, as shown on example above? As I understand, these all are the same params, they should be described similarly.

```
* `--vault_tls_client_ca=<filename>`

  Path to AcraServer TLS certificate's CA certificate for Vault certificate validation (AcraServer works as "client" when communicating with Vault).
  Empty by default.


* `--vault_tls_client_cert=<filename>`

  Path to AcraServer TLS certificate presented to Vault (AcraServer works as "client" when communicating with Vault).
  Empty by default.


* `--vault_tls_client_key=<filename>`

  Path to AcraServer TLS certificate's private key of the TLS certificate presented to Vault (AcraServer works as "client" when communicating with Vault).
  Empty by default.
```

**Review by vixentael — CHANGES_REQUESTED** [review]

**Review by vixentael — APPROVED** [review]
fantastic, thank you!

if @Lagovas doesn't have more comments, let's merge?

shall we merge it already to stable, or wait till 0.94 official release?

**Comment by Zhaars** [comment]
> fantastic, thank you!
> 
> if @Lagovas doesn't have more comments, let's merge?
> 
> shall we merge it already to stable, or wait till 0.94 official release?

Sure, let's wait for @Lagovas. I think we can merge it as related changes are already in the master.

**Review by Lagovas — APPROVED** [review]

**Pull request merged**